### PR TITLE
Update UA logic

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205617519224075/f

### Description
Cleans up the UA logic

### Steps to test this PR

_Point at the JSON blob linked in the task (Hardcoding `PRIVACY_REMOTE_CONFIG_URL` is the easiest)._

_Upgrade_
- [x] Install from develop
- [x] Verify that the `user_agent_exceptions` table in `privacy_config.db` is populated
- [x] Checkout this branch and install
- [x] Verify that the app starts (DB upgrade was successful)
- [x] Verify that the `user_agent_exceptions`, `user_agent_states`, `user_agent_sites` and `user_agent_versions` tables have been removed from `privacy_config.db`
- [x] Verify that the `user_agent_exceptions` table in the `user_agent.db` is populated

_Legacy UA_
- [x] Go to duckduckgo.com and type “user agent"
- [x] Verify that the user agent contains “DuckDuckGo” and “Version”

It should look something like this:
`Mozilla/5.0 (Linux; Android 14) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/113.0.5672.136 Mobile DuckDuckGo/5 Safari/537.36`

_Default UA_
- [x] Go to https://whatmyuseragent.com/
- [x] Verify that the user agent does not contain “DuckDuckGo” and “Version”

It should look something like this:
`Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.5672.136 Mobile Safari/537.36`

_Exceptions_
- [x] Add “whatmyuseragent.com" to the `exceptions` in `customUserAgent`
- [x] If config URL is hardcoded, uninstall and reinstall to load the new config
- [x] Go to https://whatmyuseragent.com/
- [x] Verify that the `WebView` UA is used

It should look something like this:
`Mozilla/5.0 (Linux; Android 14; sdk_gphone64_arm64 Build/UE1A.230829.036.A1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/113.0.5672.136 Mobile Safari/537.36`

_Dev settings_
- [x] Remove the exception from https://whatmyuseragent.com/
- [x] Go to dev settings and override the user agent
- [x] Verify that the chosen agent is reflected on https://whatmyuseragent.com/

### CODE REVIEW

**Review by cmonfortep — APPROVED** [review]
LGTM

Tested and works as decribed. Tested another scenario, disabling privacy protection and expecting to behave as if whatmyuseragent.com is an exception.

Just left some questions about some details I've found.

Side note: I have the feeling that many parts of the code around the feature flag could be removed in favor of `ContributesRemoteFeature`, but will leave that part for you to consider. It can be done here or as quick win / LHF.

File: app/src/internal/java/com/duckduckgo/app/dev/settings/db/DevSettingsDataStore.kt [hunk 1]
```diff
@@ -46,7 +46,7 @@ class DevSettingsSharedPreferences @Inject constructor(private val context: Cont
     private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
 
     private fun selectedUASavedValue(): UAOverride {
-        val savedValue = preferences.getString(KEY_SELECTED_UA, null) ?: return UAOverride.DDG
+        val savedValue = preferences.getString(KEY_SELECTED_UA, null) ?: return UAOverride.DEFAULT
         return UAOverride.valueOf(savedValue)
```
[the remark this answers is not in the document]
Comment (joshliebe) [reply app/src/internal/java/com/duckduckgo/app/dev/settings/db/DevSettingsDataStore.kt]: Adressed here: https://github.com/duckduckgo/Android/pull/4205/commits/cb72231ff9939b12fe6f600e080659df34066139

File: privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/useragent/RealUserAgent.kt
[the remark this answers is not in the document]
Comment (joshliebe) [reply privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/useragent/RealUserAgent.kt:1]: Adressed here: https://github.com/duckduckgo/Android/pull/4205/commits/b4bb06459ca3c541f9441c622e010eb92011d96a

File: user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/RealUserAgent.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.duckduckgo.app.browser.UriString
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.privacy.config.api.UnprotectedTemporary
+import com.duckduckgo.user.agent.store.UserAgentRepository
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface UserAgent {
+    fun useLegacyUserAgent(url: String): Boolean
```
[the remark this answers is not in the document]
Comment (joshliebe) [reply user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/RealUserAgent.kt:32]: Addressed here: https://github.com/duckduckgo/Android/pull/4205/commits/0b022b5d09b60564f965574b2b9bab8eb6567b84

File: user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/RealUserAgent.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+@ContributesBinding(AppScope::class)
+class RealUserAgent @Inject constructor(
+    private val userAgentRepository: UserAgentRepository,
+    private val unprotectedTemporary: UnprotectedTemporary,
+) : UserAgent {
+
+    private val legacySites = listOf("duckduckgo.com")
+
+    override fun useLegacyUserAgent(url: String): Boolean {
```
[the remark this answers is not in the document]
Comment (joshliebe) [reply user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/RealUserAgent.kt:44]: Addressed here: https://github.com/duckduckgo/Android/pull/4205/commits/449f646b99bd01fd9d3d037650eaf99e682ffda3

**Comment by joshliebe** [comment]
* **#4205** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4205?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joshliebe and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/UserAgentProvider.kt [hunk 4]
```diff
@@ -166,11 +142,14 @@ class RealUserAgentProvider @Inject constructor(
 
     companion object {
         const val SPACE = " "
-        val mobilePrefix = "Mozilla/5.0 (Linux; Android ${Build.VERSION.RELEASE})"
-        const val ddgFixedMobilePrefix = "Mozilla/5.0 (Linux; Android 10; K)"
-        val desktopPrefix = "Mozilla/5.0 (X11; Linux ${System.getProperty("os.arch")})"
-        val ddgFixedDesktopPrefix = "Mozilla/5.0 (X11; Linux ${System.getProperty("os.arch")})".replace("aarch64", "x86_64")
-        val fallbackDefaultUA = "$mobilePrefix AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/96.0.4664.104 Mobile Safari/537.36"
+
+        const val mobilePrefix = "Mozilla/5.0 (Linux; Android 10; K)"
```
[the remark this answers is not in the document]
Comment (joshliebe) [reply user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/UserAgentProvider.kt:146]: It’s just a renaming thing, `ddgFixedMobilePrefix` is now the `mobilePrefix` and `ddgFixedDesktopPrefix` is now `desktopPrefix`.

`mobilePrefix` and `desktopPrefix` are the fallbacks.

File: user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/UserAgentFeatureTogglesPlugin.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import com.duckduckgo.user.agent.store.UserAgentFeatureToggleRepository
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+
+@ContributesMultibinding(AppScope::class)
+class UserAgentFeatureTogglesPlugin @Inject constructor(
+    private val userAgentFeatureToggleRepository: UserAgentFeatureToggleRepository,
+    private val appBuildConfig: AppBuildConfig,
+) : FeatureTogglesPlugin {
+    override fun isEnabled(featureName: String, defaultValue: Boolean): Boolean? {
```
[the remark this answers is not in the document]
Comment (joshliebe) [reply user-agent/user-agent-impl/src/main/java/com/duckduckgo/user/agent/impl/UserAgentFeatureTogglesPlugin.kt:31]: I checked this and all the classes that extend `PrivacyFeaturePlugin` do it this way.
I’m not comfortable making the change just in this place, we should refactor all of those classes too.
I’ll make a task.

**Pull request merged**